data_structures/arrays/permutations.py

```diff
@@ -0,0 +1,22 @@
+# All permutations:
+def permute(nums: list[int]) -> list[list[int]]:
```

Comment:
```suggestion
def permute(nums: list[int]) -> list[list[int]]:
    """
    >>> from itertools import permutations
    >>> numbers in ([], [1], [1, 2], [2, 1], [-1, -3, -2], [5.5, 3.3, 4.4, 1.1, 2.2])
    >>> all(permute(nums) == permutations(nums) for nums in numbers)
    True
    """
```

Reply:
@dhruvmanila In the `keeper` can we add a hotlink so that.

> please provide doctest for the function permute

becomes

> please provide [doctest](https://docs.python.org/3/library/doctest.html) for the function permute

Reply:
They're provided, just not in the review comments. I guess we could move them.